tcl-tk build fails on Snow Leopard

With a fresh install of Tigerbrew, I'm encountering the following build error building the [tcl-tk](https://github.com/mistydemeo/tigerbrew/blob/master/Library/Formula/tcl-tk.rb) package. This happens under Snow Leopard, and I've confirmed it does **not** happen under Tiger or Leopard.

```
gcc-4.2 -c -I/private/tmp/tcl-tk--tk20250502-92359-1xv1kvk/tk8.6.16/unix/../unix -I/private/tmp/tcl-tk--tk20250502-92359-1xv1kvk/tk8.6.16/unix/../generic -I/private/tmp/tcl-tk--tk20250502-92359-1xv1kvk/tk8.6.16/unix/../bitmaps -I/private/tmp/tcl-tk--tk20250502-92359-1xv1kvk/tk8.6.16/unix/../macosx -I/private/tmp/tcl-tk--tk20250502-92359-1xv1kvk/tk8.6.16/unix/../xlib -O2 -pipe  -arch x86_64   -Wall -Wpointer-arith -fno-common -DBUILD_tk -I/private/tmp/tcl-tk20250502-92359-belyhi/tcl8.6.16/generic -I/private/tmp/tcl-tk20250502-92359-belyhi/tcl8.6.16/unix -DPACKAGE_NAME=\"tk\" -DPACKAGE_TARNAME=\"tk\" -DPACKAGE_VERSION=\"8.6\" -DPACKAGE_STRING=\"tk\ 8.6\" -DPACKAGE_BUGREPORT=\"\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DUSE_THREAD_ALLOC=1 -D_REENTRANT=1 -D_THREAD_SAFE=1 -DHAVE_PTHREAD_ATTR_SETSTACKSIZE=1 -DHAVE_PTHREAD_ATFORK=1 -DTCL_THREADS=1 -DMODULE_SCOPE=extern\ __attribute__\(\(__visibility__\(\"hidden\"\)\)\) -DHAVE_HIDDEN=1 -DMAC_OSX_TCL=1 -DHAVE_COREFOUNDATION=1 -DTCL_CFG_DO64BIT=1 -DHAVE_CAST_TO_UNION=1 -DTCL_SHLIB_EXT=\".dylib\" -DNDEBUG=1 -DTCL_CFG_OPTIMIZED=1 -DTCL_WIDE_INT_IS_LONG=1 -DHAVE_SYS_TIME_H=1 -DTIME_WITH_SYS_TIME=1 -DHAVE_INTPTR_T=1 -DHAVE_UINTPTR_T=1 -DHAVE_PW_GECOS=1 -DHAVE_AVAILABILITYMACROS_H=1 -DHAVE_WEAK_IMPORT=1 -D_DARWIN_C_SOURCE=1 -DMAC_OSX_TK=1       -std=gnu99 -x objective-c -DUSE_TCL_STUBS /private/tmp/tcl-tk--tk20250502-92359-1xv1kvk/tk8.6.16/unix/../macosx/tkMacOSXWm.c
/private/tmp/tcl-tk--tk20250502-92359-1xv1kvk/tk8.6.16/unix/../macosx/tkMacOSXWindowEvent.c: In function ?-[TKContentView(TKWindowEvent) initWithFrame:]?:
/private/tmp/tcl-tk--tk20250502-92359-1xv1kvk/tk8.6.16/unix/../macosx/tkMacOSXWindowEvent.c:981: error: assignment of read-only variable ?prop.238?
make: *** [tkMacOSXWindowEvent.o] Error 1
```

I note that the error is in what looks like a `tk` library, and sure enough the package builds successfully when the `--without-tk` flag is supplied.

I'm investigating right now too, but this one's a bit outside my wheelhouse, so I thought I'd create an issue in case somebody else recognizes this.

--- a/Library/Formula/tcl-tk.rb
+++ b/Library/Formula/tcl-tk.rb
@@ -65,6 +65,12 @@ class TclTk < Formula
           args << "--without-x"
         end
 
+        # Tk Aqua build fix on 10.6 & unbreak the ability to run the test suite on < 10.8
+        # https://core.tcl-lang.org/tk/tktview/c789692418
+        # https://core.tcl-lang.org/tk/tktview/044f2bc55b
+        # Need to use this way to patch a resource as the usual method doesn't work on resources
+        system "patch", "-p0", "-i", "#{HOMEBREW_PREFIX}/Library/Formula/tcl-tk.rb"
+
         cd "unix" do
           system "./configure", *args
           system "make", "TK_LIBRARY=#{lib}"
@@ -89,3 +95,31 @@ class TclTk < Formula
     assert_equal "honk", pipe_output("#{bin}/tclsh", "puts honk\n").chomp
   end
 end
+__END__
+--- macosx/tkMacOSXWindowEvent.c.orig	2025-07-26 14:20:28.000000000 +0100
++++ macosx/tkMacOSXWindowEvent.c	2025-07-26 14:24:11.000000000 +0100
+@@ -977,8 +977,11 @@
+ 	self.layer = [CALayer layer];
+ 	self.wantsLayer = YES;
+ 	self.layerContentsRedrawPolicy = NSViewLayerContentsRedrawOnSetNeedsDisplay;
+-	self.layer.contentsGravity = self.layer.contentsAreFlipped ?
+-	    kCAGravityTopLeft : kCAGravityBottomLeft;
++	if (self.layer.contentsAreFlipped) {
++	    self.layer.contentsGravity = kCAGravityTopLeft;
++	} else {
++	    self.layer.contentsGravity = kCAGravityBottomLeft;
++	}
+ 
+ 	/*
+ 	 * Nothing gets drawn at all if the layer does not have a delegate.
+--- macosx/tkMacOSXTest.c.orig	2025-07-26 15:14:47.000000000 +0100
++++ macosx/tkMacOSXTest.c	2025-07-26 15:15:42.000000000 +0100
+@@ -88,7 +88,7 @@
+     TCL_UNUSED(void *),		/* Not used. */
+     TCL_UNUSED(Tcl_Interp *),			/* Not used. */
+     TCL_UNUSED(int),				/* Not used. */
+-    TCL_UNUSED(Tcl_Obj *const *)			/* Not used. */
++    TCL_UNUSED(Tcl_Obj *const *))			/* Not used. */
+ {
+     Debugger();
+     return TCL_OK;

